### PR TITLE
taskArraySort

### PR DESCRIPTION
**Pull request by MaxGmn**

### CODE REVIEW

File: ArraySortDemonstration/ArraySortDemonstration/ViewController.swift [hunk 1]
```diff
@@ -10,11 +10,128 @@ import UIKit
...
+                
+                if j < rightArray.count{
+                    resultArray += rightArray[j...rightArray.count - 1]
+                }
+                
+                arrayForSort.append(resultArray)
+            }
+        }
+        
+        tableView.reloadData()
```
Comment (ghost) [inline ArraySortDemonstration/ArraySortDemonstration/ViewController.swift]: implement it without using `reloadData()`

File: ArraySortDemonstration/ArraySortDemonstration/ViewController.swift [hunk 2]
```diff
@@ -10,11 +10,128 @@ import UIKit
...
+        case 2:
+            mergeSort()
+        default:
+            break
+        }
+    }
+    
+    @IBAction func changeSortType(_ sender: Any) {
+        arrayForSort = [arrayFilling()]
+        tableView.reloadData()
```
Comment (vitaliiBoiarskyi) [inline ArraySortDemonstration/ArraySortDemonstration/ViewController.swift]: implement it without using reloadData()

**Review by ghost — CHANGES_REQUESTED** [review]
see comments

**Review by ghost — CHANGES_REQUESTED** [review]
see comments

File: ArraySortDemonstration/ArraySortDemonstration/ViewController.swift [hunk 3]
```diff
@@ -10,11 +10,165 @@ import UIKit
...
+        let currentCell = tableView.dequeueReusableCell(withIdentifier: "cellID", for: indexPath)
+        currentCell.textLabel?.text = "\(arrayForSort[indexPath.section][indexPath.row])"
+        
+        return currentCell
+    }
+    
+    
+}
 
+extension ViewController: UITableViewDelegate{
```
Comment (ghost) [inline ArraySortDemonstration/ArraySortDemonstration/ViewController.swift]: Do you really need this extension?

**Review by ghost — APPROVED** [review]

**Pull request merged**